### PR TITLE
du: replace getopts with clap

### LINKED ISSUE #2120: du: move from getopts to clap
Code is in:
https://github.com/uutils/coreutils/blob/master/src/uu/du/src/du.rs

Tests:
https://github.com/uutils/coreutils/blob/master/tests/by-util/test_du.rs

As example:
https://github.com/uutils/coreutils/commit/9081e120d6b866c94679d9e29c7bdb17ccccb4d0

### PR DESCRIPTION
**Pull request by syukronrm**
Closes #2120

### CODE REVIEW

File: src/uu/du/src/du.rs
Comment (sylvestre) [inline src/uu/du/src/du.rs:51]: please rename  the one letter option by something more explicit
it is not obvious when we read options::L

**Comment by sylvestre** [comment]
besides that, looks great :)

File: src/uu/du/src/du.rs
Comment (jhscheer) [inline src/uu/du/src/du.rs:389]: there are brackets missing, the output should be
```
du [OPTION]... [FILE]...
du [OPTION]... --files0-from=F
```

Comment (syukronrm) [reply src/uu/du/src/du.rs:389]: Good catch!

File: src/uu/du/src/du.rs
Comment (ycd) [inline src/uu/du/src/du.rs:670]: Can you fix the clippy warnings too?
See: https://rust-lang.github.io/rust-clippy/master/#redundant_slicing

File: src/uu/du/src/du.rs
Comment (ycd) [inline src/uu/du/src/du.rs:619]: Ditto
See: https://rust-lang.github.io/rust-clippy/master/#redundant_slicing

File: src/uu/du/src/du.rs
Comment (ycd) [inline src/uu/du/src/du.rs:563]: Redundant clone
See: https://rust-lang.github.io/rust-clippy/master/#redundant_clone

**Review by ycd — CHANGES_REQUESTED** [review]
LGTM, it'd be great if you fix the clippy warnings too :)

File: src/uu/du/src/du.rs
Comment (syukronrm) [reply src/uu/du/src/du.rs]: Fixed :+1:

File: src/uu/du/src/du.rs
Comment (syukronrm) [reply src/uu/du/src/du.rs]: Fixed :+1:

File: src/uu/du/src/du.rs
Comment (syukronrm) [reply src/uu/du/src/du.rs]: Fixed :+1:

**Comment by syukronrm** [comment]
Updated.

File: src/uu/du/src/du.rs
Comment (sylvestre) [inline src/uu/du/src/du.rs:519]: when you get a chance, could you please open good first issues for these three options? thanks

if people are looking to new bugs to contribute

**Review by sylvestre — APPROVED** [review]
waiting for the CI results

**Comment by sylvestre** [comment]
fails on:

```

---- test_du::test_du_time stdout ----
current_directory_resolved: 
run: /target/i686-unknown-linux-musl/debug/coreutils touch -a -m -t 201505150000 date_test
run: /target/i686-unknown-linux-musl/debug/coreutils du --time date_test
thread 'test_du::test_du_time' panicked at 'Command was expected to succeed.
stdout = 
 stderr = du: error: invalid argument 'modified' for '--time'
    Valid arguments are:
      - 'accessed', 'created', 'modified'
    Try 'du --help' for more information.
', tests/common/util.rs:143:13
```

**Comment by tertsdiepraam** [comment]
The problem is that clap parses this: `du --time date_test` as setting the value for `--time` to `date_test`. This can be fixed by adding `require_equals` and `min_values`:
```rust
.arg(
    Arg::with_name(options::TIME)
        .long(options::TIME)
        .value_name("WORD")
        .require_equals(true)
        .min_values(0)
        .help(
            "show time of the last modification of any file in the \
            directory, or any of its subdirectories.  If WORD is given, show time as WORD instead \
            of modification time: atime, access, use, ctime or status"
        )
)
```

**Comment by syukronrm** [comment]
@tertsdiepraam Thanks! Works well.

**Pull request merged**
